calendar-app/Components/API/Users/getUserByUsername.ts

```diff
@@ -0,0 +1,23 @@
+import backendURL from "../BeUrl";
+
+export async function getUserByUsername(username) {
+  try {
+    const encodedUsername = encodeURIComponent(username);
+    const response = await fetch(`${backendURL}/user/get/${encodedUsername}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch user by username: ${username}`);
+    }
+
+    const user = await response.json();
+    return user;
+  } catch (error) {
+    console.error('Error fetching user:', error);
+    throw error;
+  }
+}
```

Comment:
We already have an api "UsernameCheck.ts", it does exactly the same job, it would probably be smarter to reuse it